### PR TITLE
Bump compile version to 28 and migrate to AndroidX

### PR DESCRIPTION
**Pull request by acrown-msft**
Updated Facebook SDK to be compatible with AndroidX fragments

### CODE REVIEW

File: EmbeddedSocialClient/sdk/build.gradle [hunk 1]
```diff
@@ -91,14 +91,14 @@ dependencies {
...
-    api 'com.android.support:cardview-v7:27.0.2'
-    api 'com.android.support:design:27.0.2'
-    api 'com.android.support:support-v4:27.0.2'
-    api 'com.android.support:support-annotations:27.0.2'
+    api 'androidx.appcompat:appcompat:1.0.0'
+    api 'androidx.recyclerview:recyclerview:1.0.0'
+    api 'androidx.cardview:cardview:1.0.0'
+    api 'androidx.annotation:annotation:1.0.0'
+    api 'androidx.legacy:legacy-support-v4:1.0.0'
+    api 'com.google.android.material:material:1.0.0-rc01'
```
[the remark this answers is not in the document]
Comment (acrown-msft) [reply EmbeddedSocialClient/sdk/build.gradle]: Thanks Sean.  Is there an up to date table with the most recent versions of everything? I was going off of https://developer.android.com/jetpack/androidx/migrate#artifact_mappings

[the remark this answers is not in the document]
Comment (barbeau) [reply EmbeddedSocialClient/sdk/build.gradle]: I think this is the best listing for source Google Maven repo - https://dl.google.com/dl/android/maven2/index.html.

IIRC there are three main library groups now to track version bumps:
* Google Play Services
* Android X (formerly support libraries)
* Material design

I believe Android Studio will flag the first two in build.gradle when upgrades are available.  For material design you can check Google Maven repo or GitHub - https://github.com/material-components/material-components-android.

[the remark this answers is not in the document]
Comment (barbeau) [reply EmbeddedSocialClient/sdk/build.gradle]: Oh, and Firebase, which I believe Android Studio will also flag when updates are available.  Here are the Firebase release notes which show the current versions:
* https://firebase.google.com/support/release-notes/android

[the remark this answers is not in the document]
Comment (barbeau) [reply EmbeddedSocialClient/sdk/build.gradle]: Firebase issue opened at https://github.com/Microsoft/EmbeddedSocial-Android-SDK/issues/113.

**Pull request merged**